mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +126,31 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    kwargs = {'decode_times': False,
+              'concat_dim': 'Time'}
+
+    softLimit = resource.getrlimit(resource.RLIMIT_NOFILE)[0]
+
+    # use autoclose if we will use more than half the soft limit on open files
+    autoclose = len(fileNames) > softLimit/2
+
+    try:
+        ds = xarray.open_mfdataset(fileNames,
+                                   preprocess=preprocess_partial,
+                                   autoclose=autoclose, **kwargs)
+    except TypeError as e:
+        if 'autoclose' in str(e):
+            if autoclose:
+                # This indicates that xarray version doesn't support autoclose
+                print 'Warning: open_multifile_dataset is trying to use autoclose=True but\n' \
+                      'it appears your xarray version doesn\'t support this argument. Will\n' \
+                      'try again without autoclose argument.'
+
```

Comment:
If you want to make a more specific suggestion, I'm happy to add another try/except for the error your want.  I don't understand exactly what exception I should be catching here and what the message should be, so I don't think I'm in a position to add the code you want.